test/foo/lib/config/test_configloader.py

```diff
@@ -0,0 +1,26 @@
+import pytest
+
+from foo.lib.config import ConfigLoader, ConfigLoaderException
+
+
+def test_load(monkeypatch):
+    loader = ConfigLoader(defaults={'VAR0': 'default value 0',
+                                    'VAR1': 'default value 1'})
+    monkeypatch.setenv('VAR3', 'not bar')
+    config = loader.load('test/foo/lib/config/config.yaml')
+    assert config._config == {'VAR0': 'fizz',
+                              'VAR1': 'default value 1',
+                              'VAR2': 'foo',
+                              'VAR3': 'not bar'}
+    assert config.source == 'test/foo/lib/config/config.yaml'
+
+
+def test_load_missing():
+    loader = ConfigLoader()
+    loader.reset()
```

Comment:
remove it:
The loader should always raise an exception, when missing config file, not only when the loader is in special state.